dev_docs/ARCHITECTURE.md

```diff
@@ -1,9 +1,86 @@
 # Architecture of the APM Server
 
 This document gives a high level overview over the architecture of the main APM Server components.
-The main purpose of the APM Server is to ingest data. It validates, enriches and transforms data, that it receives from a variety of APM agents, into a dedicated format and passes them on to an output, such as Elasticsearch. 
+The main purpose of the APM Server is to ingest data. It validates, enriches and transforms data,
+received from a variety of APM agents, into a dedicated format and passes them on to an output,
+such as Elasticsearch.
 
 ## Ingest Flow
-High level overview over incoming data and their flow through the APM Server until being passed on to the output publisher pipeline. 
 
-![](./docs/images/ingest-flow.png)
\ No newline at end of file
+High level overview over incoming data and their flow through the APM Server until being passed on
+to the output publisher pipeline.
+
+![ingest-flow](../docs/images/ingest-flow.png)
+
+## Modelindexer Architecture
+
+When APM Server runs under managed mode or using the Elasticsearch output in standalone mode, it uses
+a custom Elasticsearch output called `modelindexer`.  This custom output differs from the previous
+libbeat output in multiple ways, but its main difference is that it fills a local cache until it is full,
```

Comment:
Isn't it still useful for us devs to come back to this since `7.17` is LTS? I expect maybe some team members may want to look into where we came from and how that looked compared to what we have

Reply:
My main concern is that we do not have any libbeat related dev-docs that would explain how libbeat output was working, so I didn't consider this comparison very helpful. I don't feel strongly about this, if you believe it adds value for new team members, then please keep it.